### ISSUE TITLE
Pods located on separate worker nodes unable to ping/reach each other

### ISSUE THREAD

**Issue by neerajdixit-msft2** [cilium/cilium#28702]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

We installed cilium 1.14.2 via helm on a kubernetes system with 2 worker nodes. The system was functional at the start and after installing the workload pods, the pods located on different worker nodes are unable to reach each other. As a result of this connectivity loss, many of the workload pods are in CrashLoopBackOff stage because of liveness/readiness failures.

> # ping to pod located on another worker node
sh-4.4# ping 10.0.2.225
PING 10.0.2.225 (10.0.2.225) 56(84) bytes of data.
--- 10.0.2.225 ping statistics ---
**608 packets transmitted, 0 received, 100% packet loss, time 1166ms**

There is no error seen in cilium status but the cilium health check seems to be failing for 1 worker node as follows:

> PS C:\Users\ndixit> kubectl -n kube-system exec -ti cilium-z4q5l -- cilium-health status
Defaulted container "cilium-agent" out of: cilium-agent, config (init), mount-cgroup (init), apply-sysctl-overwrites (init), mount-bpf-fs (init), clean-cilium-state (init), install-cni-binaries (init)
Probe time:   2023-10-19T03:03:18Z
Nodes:
  nodepool-5455479bcf-nmjd4 (localhost):
    Host connectivity to 10.57.55.123:
      ICMP to stack:   OK, RTT=257.173µs
      HTTP to agent:   OK, RTT=216.769µs
    Endpoint connectivity to 10.0.1.252:
      ICMP to stack:   OK, RTT=223.238µs
      HTTP to agent:   OK, RTT=269.331µs
  nodepool-5455479bcf-j9m8x:
    Host connectivity to 10.57.55.124:
      ICMP to stack:   OK, RTT=437.029µs
      HTTP to agent:   OK, RTT=467.004µs
    Endpoint connectivity to 10.0.2.14:
      ICMP to stack:   OK, RTT=523.799µs
      **HTTP to agent:   Get "http://10.0.2.14:4240/hello": context deadline exceeded (Client.Timeout exceeded while awaiting headers)**
  target-cluster-control-plane-z2lf4:
    Host connectivity to 10.57.55.122:
      ICMP to stack:   OK, RTT=428.734µs
      HTTP to agent:   OK, RTT=437.487µs
    Endpoint connectivity to 10.0.0.40:
      ICMP to stack:   OK, RTT=510.344µs
      HTTP to agent:   OK, RTT=1.802989ms

### Cilium Version

1.14.2

### Kernel Version

5.15.131.1-2.cm2 #1 SMP Sun Sep 24 03:38:45 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

### Kubernetes Version

1.26.3

### Sysdump

[cilium-sysdump-20231019-124101.zip](https://github.com/cilium/cilium/files/13047086/cilium-sysdump-20231019-124101.zip)

### Relevant log output

```shell
Cilium monitor does show a lot of packet drops
-------------------------------------------------
PS> kubectl -n kube-system exec -ti cilium-z4q5l -- cilium monitor --type drop
Defaulted container "cilium-agent" out of: cilium-agent, config (init), mount-cgroup (init), apply-sysctl-overwrites (init), mount-bpf-fs (init), clean-cilium-state (init), install-cni-binaries (init)
Press Ctrl-C to quit
level=info msg="Initializing dissection cache..." subsys=monitor
xx drop (No tunnel/encapsulation endpoint (datapath BUG!)) flow 0x44044f8a to endpoint 0, ifindex 28, file bpf_host.c:1092, , identity 22883->unknown: 10.0.1.40 -> 10.0.2.84
xx drop (No tunnel/encapsulation endpoint (datapath BUG!)) flow 0x58783c3c to endpoint 0, ifindex 28, file bpf_host.c:1092, , identity 22883->unknown: 10.0.1.40 -> 10.0.2.84
xx drop (No tunnel/encapsulation endpoint (datapath BUG!)) flow 0x6a623100 to endpoint 0, ifindex 28, file bpf_host.c:1092, , identity 26171->unknown: 10.0.1.40 -> 10.0.2.84
xx drop (No tunnel/encapsulation endpoint (datapath BUG!)) flow 0xf2baca39 to endpoint 0, ifindex 28, file bpf_host.c:1092, , identity 11924->unknown: 10.0.1.40 -> 10.0.2.84
xx drop (No tunnel/encapsulation endpoint (datapath BUG!)) flow 0xa8d50074 to endpoint 0, ifindex 28, file bpf_host.c:1092, , identity 14180->unknown: 10.0.1.40 -> 10.0.2.84
xx drop (No tunnel/encapsulation endpoint (datapath BUG!)) flow 0x25160692 to endpoint 0, ifindex 28, file bpf_host.c:1092, , identity 16808->unknown: 10.0.1.40 -> 10.0.2.84
xx drop (No tunnel/encapsulation endpoint (datapath BUG!)) flow 0x283a37bb to endpoint 0, ifindex 28, file bpf_host.c:1092, , identity 8465->unknown: 10.0.1.40 -> 10.0.2.84
xx drop (No tunnel/encapsulation endpoint (datapath BUG!)) flow 0x9f9cfaa9 to endpoint 0, ifindex 28, file bpf_host.c:1092, , identity 11924->unknown: 10.0.1.40 -> 10.0.2.84
xx drop (No tunnel/encapsulation endpoint (datapath BUG!)) flow 0xec39174d to endpoint 0, ifindex 28, file bpf_host.c:1092, , identity 10533->unknown: 10.0.1.40 -> 10.0.2.84
xx drop (No tunnel/encapsulation endpoint (datapath BUG!)) flow 0x7351929f to endpoint 0, ifindex 28, file bpf_host.c:1092, , identity 28020->unknown: 10.0.1.40 -> 10.0.2.84
xx drop (No tunnel/encapsulation endpoint (datapath BUG!)) flow 0x78a690f1 to endpoint 0, ifindex 28, file bpf_host.c:1092, , identity 3091->unknown: 10.0.1.40 -> 10.0.2.84
xx drop (No tunnel/encapsulation endpoint (datapath BUG!)) flow 0xad127bc2 to endpoint 0, ifindex 28, file bpf_host.c:1092, , identity 17518->unknown: 10.0.1.40 -> 10.0.2.84
xx drop (No tunnel/encapsulation endpoint (datapath BUG!)) flow 0x1424d9b5 to endpoint 0, ifindex 28, file bpf_host.c:1092, , identity 28020->unknown: 10.0.1.40 -> 10.0.2.84
xx drop (No tunnel/encapsulation endpoint (datapath BUG!)) flow 0xbb489e2b to endpoint 0, ifindex 28, file bpf_host.c:1092, , identity 14029->unknown: 10.0.1.40 -> 10.0.2.84
xx drop (No tunnel/encapsulation endpoint (datapath BUG!)) flow 0xec64373f to endpoint 0, ifindex 28, file bpf_host.c:1092, , identity 31186->unknown: 10.0.1.40 -> 10.0.2.84
xx drop (No tunnel/encapsulation endpoint (datapath BUG!)) flow 0xd84631ee to endpoint 0, ifindex 28, file bpf_host.c:1092, , identity 11924->unknown: 10.0.1.40 -> 10.0.2.84
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by neerajdixit-msft2** [comment]
Further debugging revealed that CiliumInternalIP changed from from **10.0.2.84->10.0.2.123**. Node IP update is well received by other nodes, but xfrm policy configuration logic still has stale CiliumInternalIP.

![CiliumEndpointIpChange](https://github.com/cilium/cilium/assets/123779900/40737e13-4e99-4c4f-af15-66cd0269659b)

We also noticed that cilium is trying to update the endpoint but its failing repeatedly as follows:
`level=info msg="Successful endpoint creation" containerID=156a4a3e9a datapathPolicyRevision=1 desiredPolicyRevision=1 endpointID=727 identity=16833 ipv4=10.0.2.210 ipv6= k8sPodName=core/etcd-cluster-1-7465798c89-tp2bn subsys=daemon`

`level=warning msg="Unable to update ipcache map entry on pod update" error="no/invalid HostIP: " k8sNamespace=core k8sPodName=etcd-cluster-1-7465798c89-tp2bn new-hostIP= new-podIP= new-podIPs="[]" old-hostIP= old-podIP= old-podIPs="[]" subsys=k8s-watcher`

`level=info msg="Delete endpoint request" containerID=7311366723 endpointID=2567 k8sNamespace=core k8sPodName=operator-for-redis-57d94b545b-8qh9k subsys=daemon`

I think this is Cilium IPsec configuration bug by not handling Cilium agent IP change.

**Comment by neerajdixit-msft2** [comment]
Additional info regarding the reason for adding another/duplicate cilium node object for same node:
It seems like the lookup for ciliumnode for worker node has a conflict as shown below:

```powershell
2023-10-17T09:54:10.911300191-07:00 level=info msg="Creating or updating CiliumNode resource" node=nodepool-5455479bcf-j9m8x subsys=nodediscovery
2023-10-17T09:54:10.916027281-07:00 level=warning msg="Detected multiple IPs of the same address type and family, Cilium will only consider the first IP in the Node resource" k8sNodeID=b5f35436-a5d6-4af8-a56e-d45eed3c44b4 node=nodepool-5455479bcf-j9m8x nodeName=nodepool-5455479bcf-j9m8x subsys=k8s type=ExternalIP
2023-10-17T09:54:10.916034593-07:00 level=warning msg="Detected multiple IPs of the same address type and family, Cilium will only consider the first IP in the Node resource" k8sNodeID=b5f35436-a5d6-4af8-a56e-d45eed3c44b4 node=nodepool-5455479bcf-j9m8x nodeName=nodepool-5455479bcf-j9m8x subsys=k8s type=ExternalIP
2023-10-17T09:54:10.916083239-07:00 level=warning msg="Detected multiple IPs of the same address type and family, Cilium will only consider the first IP in the Node resource" k8sNodeID=b5f35436-a5d6-4af8-a56e-d45eed3c44b4 node=nodepool-5455479bcf-j9m8x nodeName=nodepool-5455479bcf-j9m8x subsys=k8s type=ExternalIP
2023-10-17T09:54:10.916102773-07:00 level=warning msg="Detected multiple IPs of the same address type and family, Cilium will only consider the first IP in the Node resource" k8sNodeID=b5f35436-a5d6-4af8-a56e-d45eed3c44b4 node=nodepool-5455479bcf-j9m8x nodeName=nodepool-5455479bcf-j9m8x subsys=k8s type=ExternalIP
2023-10-17T09:54:10.916109160-07:00 level=warning msg="Detected multiple IPs of the same address type and family, Cilium will only consider the first IP in the Node resource" k8sNodeID=b5f35436-a5d6-4af8-a56e-d45eed3c44b4 node=nodepool-5455479bcf-j9m8x nodeName=nodepool-5455479bcf-j9m8x subsys=k8s type=ExternalIP
2023-10-17T09:54:10.916114552-07:00 level=warning msg="Detected multiple IPs of the same address type and family, Cilium will only consider the first IP in the Node resource" k8sNodeID=b5f35436-a5d6-4af8-a56e-d45eed3c44b4 node=nodepool-5455479bcf-j9m8x nodeName=nodepool-5455479bcf-j9m8x subsys=k8s type=ExternalIP
2023-10-17T09:54:10.916119603-07:00 level=warning msg="Detected multiple IPs of the same address type and family, Cilium will only consider the first IP in the Node resource" k8sNodeID=b5f35436-a5d6-4af8-a56e-d45eed3c44b4 node=nodepool-5455479bcf-j9m8x nodeName=nodepool-5455479bcf-j9m8x subsys=k8s type=ExternalIP
2023-10-17T09:54:10.916125018-07:00 level=warning msg="Detected multiple IPs of the same address type and family, Cilium will only consider the first IP in the Node resource" k8sNodeID=b5f35436-a5d6-4af8-a56e-d45eed3c44b4 node=nodepool-5455479bcf-j9m8x nodeName=nodepool-5455479bcf-j9m8x subsys=k8s type=ExternalIP
2023-10-17T09:54:10.916143723-07:00 level=warning msg="Detected multiple IPs of the same address type and family, Cilium will only consider the first IP in the Node resource" k8sNodeID=b5f35436-a5d6-4af8-a56e-d45eed3c44b4 node=nodepool-5455479bcf-j9m8x nodeName=nodepool-5455479bcf-j9m8x subsys=k8s type=ExternalIP
2023-10-17T09:54:10.916155886-07:00 level=warning msg="Detected multiple IPs of the same address type and family, Cilium will only consider the first IP in the Node resource" k8sNodeID=b5f35436-a5d6-4af8-a56e-d45eed3c44b4 node=nodepool-5455479bcf-j9m8x nodeName=nodepool-5455479bcf-j9m8x subsys=k8s type=ExternalIP
2023-10-17T09:54:10.916169884-07:00 level=warning msg="Detected multiple IPs of the same address type and family, Cilium will only consider the first IP in the Node resource" k8sNodeID=b5f35436-a5d6-4af8-a56e-d45eed3c44b4 node=nodepool-5455479bcf-j9m8x nodeName=nodepool-5455479bcf-j9m8x subsys=k8s type=ExternalIP
2023-10-17T09:54:10.916262936-07:00 level=warning msg="Detected multiple IPs of the same address type and family, Cilium will only consider the first IP in the Node resource" k8sNodeID=b5f35436-a5d6-4af8-a56e-d45eed3c44b4 node=nodepool-5455479bcf-j9m8x nodeName=nodepool-5455479bcf-j9m8x subsys=k8s type=ExternalIP
2023-10-17T09:54:10.916276378-07:00 level=warning msg="Detected multiple IPs of the same address type and family, Cilium will only consider the first IP in the Node resource" k8sNodeID=b5f35436-a5d6-4af8-a56e-d45eed3c44b4 node=nodepool-5455479bcf-j9m8x nodeName=nodepool-5455479bcf-j9m8x subsys=k8s type=ExternalIP
2023-10-17T09:54:10.916283182-07:00 level=warning msg="Detected multiple IPs of the same address type and family, Cilium will only consider the first IP in the Node resource" k8sNodeID=b5f35436-a5d6-4af8-a56e-d45eed3c44b4 node=nodepool-5455479bcf-j9m8x nodeName=nodepool-5455479bcf-j9m8x subsys=k8s type=ExternalIP
2023-10-17T09:54:10.916306724-07:00 level=warning msg="Detected multiple IPs of the same address type and family, Cilium will only consider the first IP in the Node resource" k8sNodeID=b5f35436-a5d6-4af8-a56e-d45eed3c44b4 node=nodepool-5455479bcf-j9m8x nodeName=nodepool-5455479bcf-j9m8x subsys=k8s type=ExternalIP
2023-10-17T09:54:10.916319991-07:00 level=warning msg="Detected multiple IPs of the same address type and family, Cilium will only consider the first IP in the Node resource" k8sNodeID=b5f35436-a5d6-4af8-a56e-d45eed3c44b4 node=nodepool-5455479bcf-j9m8x nodeName=nodepool-5455479bcf-j9m8x subsys=k8s type=ExternalIP
2023-10-17T09:54:10.916325774-07:00 level=warning msg="Detected multiple IPs of the same address type and family, Cilium will only consider the first IP in the Node resource" k8sNodeID=b5f35436-a5d6-4af8-a56e-d45eed3c44b4 node=nodepool-5455479bcf-j9m8x nodeName=nodepool-5455479bcf-j9m8x subsys=k8s type=ExternalIP
2023-10-17T09:54:10.916329176-07:00 level=warning msg="Detected multiple IPs of the same address type and family, Cilium will only consider the first IP in the Node resource" k8sNodeID=b5f35436-a5d6-4af8-a56e-d45eed3c44b4 node=nodepool-5455479bcf-j9m8x nodeName=nodepool-5455479bcf-j9m8x subsys=k8s type=ExternalIP
2023-10-17T09:54:10.916332455-07:00 level=warning msg="Detected multiple IPs of the same address type and family, Cilium will only consider the first IP in the Node resource" k8sNodeID=b5f35436-a5d6-4af8-a56e-d45eed3c44b4 node=nodepool-5455479bcf-j9m8x nodeName=nodepool-5455479bcf-j9m8x subsys=k8s type=ExternalIP
2023-10-17T09:54:10.916336725-07:00 level=warning msg="Detected multiple IPs of the same address type and family, Cilium will only consider the first IP in the Node resource" k8sNodeID=b5f35436-a5d6-4af8-a56e-d45eed3c44b4 node=nodepool-5455479bcf-j9m8x nodeName=nodepool-5455479bcf-j9m8x subsys=k8s type=ExternalIP
2023-10-17T09:54:10.916341951-07:00 level=warning msg="Detected multiple IPs of the same address type and family, Cilium will only consider the first IP in the Node resource" k8sNodeID=b5f35436-a5d6-4af8-a56e-d45eed3c44b4 node=nodepool-5455479bcf-j9m8x nodeName=nodepool-5455479bcf-j9m8x subsys=k8s type=ExternalIP
2023-10-17T09:54:10.916376024-07:00 level=warning msg="Detected multiple IPs of the same address type and family, Cilium will only consider the first IP in the Node resource" k8sNodeID=b5f35436-a5d6-4af8-a56e-d45eed3c44b4 node=nodepool-5455479bcf-j9m8x nodeName=nodepool-5455479bcf-j9m8x subsys=k8s type=ExternalIP
=======> 2023-10-17T09:54:10.935807400-07:00 level=warning msg="Unable to update CiliumNode resource, will retry" error="Operation cannot be fulfilled on ciliumnodes.cilium.io \"nodepool-5455479bcf-j9m8x\": the object has been modified; please apply your changes to the latest version and try again" subsys=nodediscovery
2023-10-17T09:54:10.939939201-07:00 level=warning msg="Detected multiple IPs of the same address type and family, Cilium will only consider the first IP in the Node resource" k8sNodeID=b5f35436-a5d6-4af8-a56e-d45eed3c44b4 node=nodepool-5455479bcf-j9m8x nodeName=nodepool-5455479bcf-j9m8x subsys=k8s type=ExternalIP
2023-10-17T09:54:10.939952512-07:00 level=warning msg="Detected multiple IPs of the same address type and family, Cilium will only consider the first IP in the Node resource" k8sNodeID=b5f35436-a5d6-4af8-a56e-d45eed3c44b4 node=nodepool-5455479bcf-j9m8x nodeName=nodepool-5455479bcf-j9m8x subsys=k8s type=ExternalIP
2023-10-17T09:54:10.939956184-07:00 level=warning msg="Detected multiple IPs of the same address type and family, Cilium will only consider the first IP in the Node resource" k8sNodeID=b5f35436-a5d6-4af8-a56e-d45eed3c44b4 node=nodepool-5455479bcf-j9m8x nodeName=nodepool-5455479bcf-j9m8x subsys=k8s type=ExternalIP
2023-10-17T09:54:10.939959508-07:00 level=warning msg="Detected multiple IPs of the same address type and family, Cilium will only consider the first IP in the Node resource" k8sNodeID=b5f35436-a5d6-4af8-a56e-d45eed3c44b4 node=nodepool-5455479bcf-j9m8x nodeName=nodepool-5455479bcf-j9m8x subsys=k8s type=ExternalIP
2023-10-17T09:54:10.939987295-07:00 level=warning msg="Detected multiple IPs of the same address type and family, Cilium will only consider the first IP in the Node resource" k8sNodeID=b5f35436-a5d6-4af8-a56e-d45eed3c44b4 node=nodepool-5455479bcf-j9m8x nodeName=nodepool-5455479bcf-j9m8x subsys=k8s type=ExternalIP
2023-10-17T09:54:10.940005223-07:00 level=warning msg="Detected multiple IPs of the same address type and family, Cilium will only consider the first IP in the Node resource" k8sNodeID=b5f35436-a5d6-4af8-a56e-d45eed3c44b4 node=nodepool-5455479bcf-j9m8x nodeName=nodepool-5455479bcf-j9m8x subsys=k8s type=ExternalIP
2023-10-17T09:54:10.940010824-07:00 level=warning msg="Detected multiple IPs of the same address type and family, Cilium will only consider the first IP in the Node resource" k8sNodeID=b5f35436-a5d6-4af8-a56e-d45eed3c44b4 node=nodepool-5455479bcf-j9m8x nodeName=nodepool-5455479bcf-j9m8x subsys=k8s type=ExternalIP
2023-10-17T09:54:10.940016139-07:00 level=warning msg="Detected multiple IPs of the same address type and family, Cilium will only consider the first IP in the Node resource" k8sNodeID=b5f35436-a5d6-4af8-a56e-d45eed3c44b4 node=nodepool-5455479bcf-j9m8x nodeName=nodepool-5455479bcf-j9m8x subsys=k8s type=ExternalIP
2023-10-17T09:54:10.940034816-07:00 level=warning msg="Detected multiple IPs of the same address type and family, Cilium will only consider the first IP in the Node resource" k8sNodeID=b5f35436-a5d6-4af8-a56e-d45eed3c44b4 node=nodepool-5455479bcf-j9m8x nodeName=nodepool-5455479bcf-j9m8x subsys=k8s type=ExternalIP
2023-10-17T09:54:10.940038669-07:00 level=warning msg="Detected multiple IPs of the same address type and family, Cilium will only consider the first IP in the Node resource" k8sNodeID=b5f35436-a5d6-4af8-a56e-d45eed3c44b4 node=nodepool-5455479bcf-j9m8x nodeName=nodepool-5455479bcf-j9m8x subsys=k8s type=ExternalIP
2023-10-17T09:54:10.940042043-07:00 level=warning msg="Detected multiple IPs of the same address type and family, Cilium will only consider the first IP in the Node resource" k8sNodeID=b5f35436-a5d6-4af8-a56e-d45eed3c44b4 node=nodepool-5455479bcf-j9m8x nodeName=nodepool-5455479bcf-j9m8x subsys=k8s type=ExternalIP
2023-10-17T09:54:10.951160873-07:00 level=info msg="Retrieved node information from cilium node" nodeName=nodepool-5455479bcf-j9m8x subsys=k8s
2023-10-17T09:54:10.951192415-07:00 level=info msg="Received own node information from API server" ipAddr.ipv4=10.57.55.124 ipAddr.ipv6="<nil>" k8sNodeIP=10.57.55.124 labels="map[beta.kubernetes.io/arch:amd64 beta.kubernetes.io/instance-type:MOCVirtualMachine beta.kubernetes.io/os:linux kubernetes.io/arch:amd64 kubernetes.io/hostname:nodepool-5455479bcf-j9m8x kubernetes.io/os:linux node.kubernetes.io/instance-type:MOCVirtualMachine topology.disk.csi.azure.com/zone: topology.kubernetes.io/zone:DZHRR33]" nodeName=nodepool-5455479bcf-j9m8x subsys=k8s v4Prefix=10.0.2.0/24 v6Prefix="<nil>"
2023-10-17T09:54:10.951200811-07:00 level=info msg="k8s mode: Allowing localhost to reach local endpoints" subsys=daemon
2023-10-17T09:54:10.952061509-07:00 level=info msg="Detected devices" devices="[]" subsys=linux-datapath
2023-10-17T09:54:10.952069846-07:00 level=info msg="Enabling k8s event listener" subsys=k8s-watcher
2023-10-17T09:54:10.952398922-07:00 level=info msg="Removing stale endpoint interfaces" subsys=daemon
2023-10-17T09:54:10.952674938-07:00 level=info msg="Waiting until local node addressing before starting watchers depending on it" subsys=k8s-watcher
2023-10-17T09:54:10.953196760-07:00 level=info msg="Skipping kvstore configuration" subsys=daemon
2023-10-17T09:54:10.953356992-07:00 level=info msg="Initializing node addressing" subsys=daemon
2023-10-17T09:54:10.953402365-07:00 level=info msg="Initializing cluster-pool IPAM" subsys=ipam v4Prefix=10.0.2.0/24 v6Prefix="<nil>"
2023-10-17T09:54:10.954246570-07:00 level=info msg="Restoring endpoints..." subsys=daemon
2023-10-17T09:54:10.954485522-07:00 level=info msg="Endpoints restored" failed=0 restored=0 subsys=daemon
2023-10-17T09:54:10.954577799-07:00 level=info msg="Addressing information:" subsys=daemon
2023-10-17T09:54:10.954595395-07:00 level=info msg="  Cluster-Name: default" subsys=daemon
2023-10-17T09:54:10.954601580-07:00 level=info msg="  Cluster-ID: 0" subsys=daemon
2023-10-17T09:54:10.954608472-07:00 level=info msg="  Local node-name: nodepool-5455479bcf-j9m8x" subsys=daemon
2023-10-17T09:54:10.954698276-07:00 level=info msg="  Node-IPv6: <nil>" subsys=daemon
2023-10-17T09:54:10.954708476-07:00 level=info msg="  External-Node IPv4: 10.57.55.124" subsys=daemon
2023-10-17T09:54:10.954756786-07:00 level=info msg="  Internal-Node IPv4: 10.0.2.123" subsys=daemon
2023-10-17T09:54:10.954762332-07:00 level=info msg="  IPv4 allocation prefix: 10.0.2.0/24" subsys=daemon
2023-10-17T09:54:10.954773692-07:00 level=info msg="  Loopback IPv4: 169.254.42.1" subsys=daemon
2023-10-17T09:54:10.954928295-07:00 level=info msg="  Local IPv4 addresses:" subsys=daemon
2023-10-17T09:54:10.954940239-07:00 level=info msg="  - 10.57.55.124" subsys=daemon
2023-10-17T09:54:10.954945914-07:00 level=info msg="  - 172.0.6.0" subsys=daemon
2023-10-17T09:54:10.955649970-07:00 level=info msg="Adding local node to cluster" node="{nodepool-5455479bcf-j9m8x default [{ExternalIP 172.0.6.0} {InternalIP 10.57.55.124} {CiliumInternalIP 10.0.2.123} {ExternalIP 172.0.6.0}] 10.0.2.0/24 [] <nil> [] 10.0.2.105 <nil> <nil> <nil> 0 local 1 map[beta.kubernetes.io/arch:amd64 beta.kubernetes.io/instance-type:MOCVirtualMachine beta.kubernetes.io/os:linux kubernetes.io/arch:amd64 kubernetes.io/hostname:nodepool-5455479bcf-j9m8x kubernetes.io/os:linux node.kubernetes.io/instance-type:MOCVirtualMachine topology.disk.csi.azure.com/zone: topology.kubernetes.io/zone:DZHRR33] map[] 1 }" subsys=nodediscovery
``` 

This error/symptom looks very similar to the issue fixed by pull request #27963 
Can anyone confirm whether the reasoning is correct?
